src/plugins/gmx/gmxplugin.cpp

```diff
@@ -51,41 +51,40 @@ bool GmxPlugin::write(const Map *map, const QString &fileName)
     stream.setAutoFormatting(true);
     stream.setAutoFormattingIndent(2);
 
-    stream.writeComment("This Document is generated by GameMaker, if you edit it by hand then you do so at your own risk!");
+    stream.writeComment("This Document is generated by Tiled, if you edit it by hand then you do so at your own risk!");
 
     stream.writeStartElement("room");
-    stream.writeStartElement("caption");
-    stream.writeEndElement();
 
     stream.writeTextElement("width", QString::number(map->tileWidth() * map->width()));
-
     stream.writeTextElement("height", QString::number(map->tileHeight() * map->height()));
 
-    stream.writeTextElement("vsnap", QString::number(map->tileHeight()));
-    stream.writeTextElement("hsnap", QString::number(map->tileWidth()));
-
     stream.writeTextElement("isometric", QString::number((map->orientation() == Map::Orientation::Isometric ? 1 : 0)));
 
     stream.writeStartElement("instances");
 
-    uint objectId = 0u;
-    foreach (const Layer *layer, map->layers()) {
+    for (const Layer *layer : map->layers()) {
 
         if (layer->layerType() != Layer::ObjectGroupType) continue;
         const ObjectGroup *objectLayer = static_cast<const ObjectGroup*>(layer);
 
-        foreach(const MapObject *object, objectLayer->objects()) {
+        bool rotating = false;
+        for (const MapObject *object : objectLayer->objects()) {
+
+            rotating = (object->rotation() != 0);
+
             stream.writeStartElement("instance");
 
             stream.writeAttribute("objName", object->name());
             stream.writeAttribute("x", QString::number((int)(object->x())));
-            stream.writeAttribute("y", QString::number((int)(object->y() - object->height())));
+            stream.writeAttribute("y", QString::number((int)(object->y() - object->height()) + (rotating ? object->height() : 0)));
 
-            stream.writeAttribute("id", QString::number(++objectId));
+            stream.writeAttribute("scaleY", QString::number(rotating ? -1 : 1));
```

Comment:
What is the idea behind inverting the Y axis when an object is rotated? Doesn't that suddenly flip the entire graphic around?